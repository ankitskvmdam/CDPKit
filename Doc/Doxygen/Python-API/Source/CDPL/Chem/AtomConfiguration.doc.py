#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Provides constants that are used to specify the <em>R/S</em> configuration of atoms.
# 
class AtomConfiguration(Boost.Python.instance):

    ##
    # \brief Specifies that the configuration of the atom is undefined.
    # 
    UNDEF = 0

    ##
    # \brief Specifies that the atom is not a stereogenic center and thus cannot be assigned a configuration.
    # 
    NONE = 1

    ##
    # \brief Specifies that the atom has <em>R</em> configuration.
    # 
    R = 2

    ##
    # \brief Specifies that the atom has <em>S</em> configuration.
    # 
    S = 4

    ##
    # \brief Specifies that the atom is a stereogenic center but has no defined configuration.
    # 
    EITHER = 8

    ##
    # \brief Specifies that the atom has <em>r</em> configuration.
    # 
    r = 16

    ##
    # \brief Specifies that the atom has <em>s</em> configuration.
    # 
    s = 17

    ##
    # \brief Specifies that the atom has <em>M</em> configuration.
    # 
    M = 19

    ##
    # \brief Specifies that the atom has <em>P</em> configuration.
    # 
    P = 18

    ##
    # \brief Specifies that the atom has <em>m</em> configuration.
    # 
    m = 21

    ##
    # \brief Specifies that the atom has <em>p</em> configuration.
    # 
    p = 20

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>square planar</em>.
    # 
    SP = 22

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>trigonal bipyramidal</em>.
    # 
    TB = 26

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>octahedral</em>.
    # 
    OH = 47
