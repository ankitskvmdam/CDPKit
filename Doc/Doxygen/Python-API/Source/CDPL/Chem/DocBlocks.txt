#
# --- Unresolved property getter/setter functions ---
#
