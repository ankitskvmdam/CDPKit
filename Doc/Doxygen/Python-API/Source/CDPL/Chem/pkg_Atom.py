#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class Atom(AtomContainer, BondContainer, Entity3D):

    ##
    # \brief 
    #
    class AtomSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Atom: pass

        ##
        # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
        # \param atom The value to test for membership.
        # \return The result of the membership test operation.
        #
        def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief 
    #
    class BondSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Bond: pass

        ##
        # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
        # \param bond The value to test for membership.
        # \return The result of the membership test operation.
        #
        def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Initializes the \e %Atom instance.
    #
    def __init__() -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getMolecule() -> Molecule: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def findBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief 
    # \return 
    #
    def getIndex() -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %Atom instance \a atom.
    # \param atom The \e %Atom instance to copy.
    # \return The assignment target \a self.
    #
    def assign(atom: Atom) -> Atom: pass

    ##
    # \brief 
    # \return 
    #
    def getAtoms() -> object: pass

    ##
    # \brief 
    # \return 
    #
    def getBonds() -> object: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getAtom(idx: int) -> Atom: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def containsAtom(atom: Atom) -> bool: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def containsAtom(arg2: Atom) -> None: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getAtomIndex(atom: Atom) -> int: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def getAtomIndex(arg2: Atom) -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getNumAtoms() -> int: pass

    ##
    # \brief 
    # \param func 
    #
    def orderAtoms(func: CDPL.ForceField.InteractionFilterFunction2) -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getEntity(idx: int) -> Entity3D: pass

    ##
    # \brief 
    # \return 
    #
    def getNumEntities() -> int: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getBond(idx: int) -> Bond: pass

    ##
    # \brief 
    # \param bond 
    # \return 
    #
    def containsBond(bond: Bond) -> bool: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def containsBond(arg2: Bond) -> None: pass

    ##
    # \brief 
    # \param func 
    #
    def orderBonds(func: BoolBond2Functor) -> None: pass

    ##
    # \brief 
    # \param bond 
    # \return 
    #
    def getBondIndex(bond: Bond) -> int: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def getBondIndex(arg2: Bond) -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getNumBonds() -> int: pass

    ##
    # \brief 
    # \param key 
    # \param value 
    #
    def setProperty(key: CDPL.Base.LookupKey, value: CDPL.Base.Variant) -> None: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def removeProperty(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \param key 
    # \param throw_ 
    # \return 
    #
    def getProperty(key: CDPL.Base.LookupKey, throw_: bool = False) -> CDPL.Base.Variant: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def isPropertySet(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    #
    def clearProperties() -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def addProperties(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def copyProperties(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def swap(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
    # \param atom The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
    # \param bond The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>key in self</tt>.
    # \param key The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __getitem__(key: CDPL.Base.LookupKey) -> CDPL.Base.Variant: pass

    ##
    # \brief 
    # \param key 
    # \param value 
    #
    def __setitem__(key: CDPL.Base.LookupKey, value: CDPL.Base.Variant) -> None: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __delitem__(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \return 
    #
    def __len__() -> int: pass

    ##
    # \brief 
    #
    molecule = property(getMolecule)

    ##
    # \brief 
    #
    index = property(getIndex)

    ##
    # \brief 
    #
    atoms = property(getAtoms)

    ##
    # \brief 
    #
    bonds = property(getBonds)
