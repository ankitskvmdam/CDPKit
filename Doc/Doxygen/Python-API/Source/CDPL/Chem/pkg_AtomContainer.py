#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class AtomContainer(Entity3DContainer):

    ##
    # \brief Initializes the \e %AtomContainer instance.
    #
    def __init__() -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getAtom(idx: int) -> Atom: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def containsAtom(atom: Atom) -> bool: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getAtomIndex(atom: Atom) -> int: pass

    ##
    # \brief 
    # \return 
    #
    def getNumAtoms() -> int: pass

    ##
    # \brief 
    # \param func 
    #
    def orderAtoms(func: CDPL.ForceField.InteractionFilterFunction2) -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getEntity(idx: int) -> Entity3D: pass

    ##
    # \brief 
    # \return 
    #
    def getNumEntities() -> int: pass

    ##
    # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
    # \param atom The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def __getitem__(idx: int) -> Atom: pass

    ##
    # \brief 
    # \return 
    #
    def __len__() -> int: pass

    ##
    # \brief 
    #
    numAtoms = property(getNumAtoms)
