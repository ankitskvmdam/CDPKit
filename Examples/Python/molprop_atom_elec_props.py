#!/bin/env python

##
# molprop_atom_elec_props.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003-2022 Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


import sys
import os

import CDPL.Chem as Chem
import CDPL.MolProp as MolProp


# function called for read molecule
def procMolecule(mol: Chem.Molecule) -> None:
    Chem.calcImplicitHydrogenCounts(mol, False)  # calculate implicit hydrogen counts and set corresponding property for all atoms
    Chem.perceiveHybridizationStates(mol, False) # perceive atom hybridization states and set corresponding property for all atoms
    Chem.perceiveSSSR(mol, False)                # perceive smallest set of smallest rings and store as Chem.MolecularGraph property
    Chem.setRingFlags(mol, False)                # perceive cycles and set corresponding atom and bond properties
    Chem.setAromaticityFlags(mol, False)         # perceive aromaticity and set corresponding atom and bond properties
    Chem.perceivePiElectronSystems(mol, False)   # perceive pi electron systems and store info as Chem.MolecularGraph property
                                                 # (required for MHMO calculations)

    # calculate sigma charges and electronegativities using the PEOE method and store values as atom properties
    # (prerequisite for MHMO calculations)
    MolProp.calcPEOEProperties(mol, False)  

    # calculate pi charges, electronegativities and other properties by a modified Hueckel MO method and store values as properties
    MolProp.calcMHMOProperties(mol, False)
                
    for atom in mol.atoms:
        print('- Atom #%s' % str(atom.getIndex()))
        print('\tSigma charge: %s' % str(MolProp.getPEOESigmaCharge(atom)))
        print('\tPi charge: %s' % str(MolProp.getMHMOPiCharge(atom)))
        print('\tTotal partial charge: %s' % str(MolProp.calcTotalPartialCharge(atom)))
        print('\tLone-pair electronegativity: %s' % str(MolProp.calcLonePairElectronegativity(atom, mol)))
        print('\tPi electronegativity: %s' % str(MolProp.calcPiElectronegativity(atom, mol)))
        print('\tSigma electronegativity: %s' % str(MolProp.getPEOESigmaElectronegativity(atom)))
        print('\tExerted inductive effect: %s' % str(MolProp.calcInductiveEffect(atom, mol)))
        print('\tFree valence electron count: %s' % str(MolProp.calcFreeValenceElectronCount(atom, mol)))
        print('\tValence electron count: %s' % str(MolProp.calcValenceElectronCount(atom)))
        
def getReaderByFileExt(filename: str) -> Chem.MoleculeReader:
    # get the extension of the input file
    name_and_ext = os.path.splitext(filename)

    if name_and_ext[1] == '':
        sys.exit('Error: could not determine input file format (file extension missing).')

    # get input handler for the format specified by the input file's extension
    ipt_handler = Chem.MoleculeIOManager.getInputHandlerByFileExtension(name_and_ext[1][1:].lower())

    if not ipt_handler:
        sys.exit('Error: unknown input file format \'%s\'' % name_and_ext[1])

    # return file reader instance
    return ipt_handler.createReader(filename)
    
def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <input mol. file>' % sys.argv[0])

    # if the input molecules are expected to be in a specific format, a reader for this format could be created directly, e.g.
    # reader = Chem.FileSDFMoleculeReader(sys.argv[1])
    reader = getReaderByFileExt(sys.argv[1]) 
    
    # create an instance of the default implementation of the Chem.Molecule interface
    mol = Chem.BasicMolecule()
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        while reader.read(mol):
            try:
                procMolecule(mol)
            except Exception as e:
                sys.exit('Error: processing of molecule failed:\n' + str(e))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule failed:\n' + str(e))

    sys.exit(0)
        
if __name__ == '__main__':
    main()
