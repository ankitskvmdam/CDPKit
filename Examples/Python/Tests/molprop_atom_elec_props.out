- Atom #0
	Sigma charge: -0.20548818062530538
	Pi charge: 0.03393007539706483
	Total partial charge: -0.17155810522824055
	Lone-pair electronegativity: 4.933807319031711
	Pi electronegativity: 6.317790455146629
	Sigma electronegativity: 11.911529354729694
	Exerted inductive effect: 24.99254458293381
	Free valence electron count: 6
	Valence electron count: 7
- Atom #1
	Sigma charge: 0.12342233600262448
	Pi charge: 0.01694267183068121
	Total partial charge: 0.1403650078333057
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.745364929854271
	Sigma electronegativity: 9.96329811181122
	Exerted inductive effect: 51.299952112336626
	Free valence electron count: 0
	Valence electron count: 4
- Atom #2
	Sigma charge: -0.026094046678615854
	Pi charge: -0.021911952716225258
	Total partial charge: -0.04800599939484111
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.414502636325802
	Sigma electronegativity: 8.547831642856119
	Exerted inductive effect: 51.342383821652696
	Free valence electron count: 0
	Valence electron count: 4
- Atom #3
	Sigma charge: -0.05280657267329697
	Pi charge: 0.004919206898790107
	Total partial charge: -0.04788736577450686
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.642031978720428
	Sigma electronegativity: 8.302053429202296
	Exerted inductive effect: 49.39056539475531
	Free valence electron count: 0
	Valence electron count: 4
- Atom #4
	Sigma charge: -0.011273328237591532
	Pi charge: -0.016887255592876738
	Total partial charge: -0.02816058383046827
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.456790137233051
	Sigma electronegativity: 8.68512448359927
	Exerted inductive effect: 42.731683140370706
	Free valence electron count: 0
	Valence electron count: 4
- Atom #5
	Sigma charge: 0.12007157434526296
	Pi charge: 0.0
	Total partial charge: 0.12007157434526296
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 9.109361356465124
	Exerted inductive effect: 48.31965671156107
	Free valence electron count: 0
	Valence electron count: 4
- Atom #6
	Sigma charge: -0.3595147962031278
	Pi charge: 0.0
	Total partial charge: -0.3595147962031278
	Lone-pair electronegativity: 3.4889301955613234
	Pi electronegativity: 3.4889301955613234
	Sigma electronegativity: 9.714727568333265
	Exerted inductive effect: 42.06225910674755
	Free valence electron count: 4
	Valence electron count: 6
- Atom #7
	Sigma charge: 0.07466227677700811
	Pi charge: 0.0
	Total partial charge: 0.07466227677700811
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 8.675879677291165
	Exerted inductive effect: 46.350583094331355
	Free valence electron count: 0
	Valence electron count: 4
- Atom #8
	Sigma charge: -0.014432098904598798
	Pi charge: -0.0015994791521685148
	Total partial charge: -0.01603157805676731
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.586363964332663
	Sigma electronegativity: 8.655807349282114
	Exerted inductive effect: 60.29158937711367
	Free valence electron count: 0
	Valence electron count: 4
- Atom #9
	Sigma charge: -0.00602395318314408
	Pi charge: 0.016728859789354456
	Total partial charge: 0.010704906606210375
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.743519946967758
	Sigma electronegativity: 8.733911551231145
	Exerted inductive effect: 56.3359746499138
	Free valence electron count: 0
	Valence electron count: 4
- Atom #10
	Sigma charge: -0.05396901594431603
	Pi charge: 0.002528784784651905
	Total partial charge: -0.051440231159664125
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.6215893347853925
	Sigma electronegativity: 8.29140687996879
	Exerted inductive effect: 63.460875525256085
	Free valence electron count: 0
	Valence electron count: 4
- Atom #11
	Sigma charge: -0.043976134596747436
	Pi charge: 0.02423308572156946
	Total partial charge: -0.019743048875177978
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.808434713989138
	Sigma electronegativity: 8.383062615183562
	Exerted inductive effect: 65.951736302312
	Free valence electron count: 0
	Valence electron count: 4
- Atom #12
	Sigma charge: 0.05212304968118334
	Pi charge: -0.017849716106740132
	Total partial charge: 0.034273333574443204
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.448678641962874
	Sigma electronegativity: 9.27988920961381
	Exerted inductive effect: 51.98888868001857
	Free valence electron count: 0
	Valence electron count: 4
- Atom #13
	Sigma charge: -0.03862460138419188
	Pi charge: 0.023494068184441774
	Total partial charge: -0.015130533199750107
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.802027197058463
	Sigma electronegativity: 8.432271423445783
	Exerted inductive effect: 66.26925197589937
	Free valence electron count: 0
	Valence electron count: 4
- Atom #14
	Sigma charge: 0.0633873906420294
	Pi charge: 0.0
	Total partial charge: 0.0633873906420294
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.563287259282516
	Exerted inductive effect: 37.59511963795683
	Free valence electron count: 0
	Valence electron count: 1
- Atom #15
	Sigma charge: 0.09916316453581474
	Pi charge: 0.25429577119672053
	Total partial charge: 0.3534589357325353
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 7.959261965698718
	Sigma electronegativity: 11.334270238100002
	Exerted inductive effect: 42.953274627865824
	Free valence electron count: 0
	Valence electron count: 4
- Atom #16
	Sigma charge: -0.19086904025381626
	Pi charge: -0.30183137441782915
	Total partial charge: -0.4927004146716454
	Lone-pair electronegativity: 2.1689103824540297
	Pi electronegativity: 6.256423543731014
	Sigma electronegativity: 13.436995861587947
	Exerted inductive effect: 52.8770660312853
	Free valence electron count: 2
	Valence electron count: 5
- Atom #17
	Sigma charge: 0.06304302305316742
	Pi charge: 0.0
	Total partial charge: 0.06304302305316742
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.561162787108583
	Exerted inductive effect: 36.81131404190891
	Free valence electron count: 0
	Valence electron count: 1
- Atom #18
	Sigma charge: 0.06217610963025851
	Pi charge: 0.0
	Total partial charge: 0.06217610963025851
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.555814037671911
	Exerted inductive effect: 36.63164130977531
	Free valence electron count: 0
	Valence electron count: 1
- Atom #19
	Sigma charge: 0.06104476307207405
	Pi charge: 0.0
	Total partial charge: 0.06104476307207405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.548832502234568
	Exerted inductive effect: 28.276581921357057
	Free valence electron count: 0
	Valence electron count: 1
- Atom #20
	Sigma charge: 0.06104476307207405
	Pi charge: 0.0
	Total partial charge: 0.06104476307207405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.548832502234568
	Exerted inductive effect: 28.276581921357057
	Free valence electron count: 0
	Valence electron count: 1
- Atom #21
	Sigma charge: -0.014152656523774019
	Pi charge: 0.0
	Total partial charge: -0.014152656523774019
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.850455172762714
	Exerted inductive effect: 30.330320754964674
	Free valence electron count: 0
	Valence electron count: 4
- Atom #22
	Sigma charge: -0.03697795804610718
	Pi charge: 0.0
	Total partial charge: -0.03697795804610718
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.643112999573505
	Exerted inductive effect: 18.27431774085019
	Free valence electron count: 0
	Valence electron count: 4
- Atom #23
	Sigma charge: -0.0015796771267166405
	Pi charge: 0.0
	Total partial charge: -0.0015796771267166405
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.9655032552908125
	Exerted inductive effect: 11.146592380047533
	Free valence electron count: 0
	Valence electron count: 4
- Atom #24
	Sigma charge: -0.3083406547856823
	Pi charge: 0.0
	Total partial charge: -0.3083406547856823
	Lone-pair electronegativity: 1.5790212170034237
	Pi electronegativity: 1.5790212170034237
	Sigma electronegativity: 8.3330546999943
	Exerted inductive effect: 6.653701868042446
	Free valence electron count: 2
	Valence electron count: 5
- Atom #25
	Sigma charge: -0.013050397513689383
	Pi charge: 0.0
	Total partial charge: -0.013050397513689383
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.86051753902983
	Exerted inductive effect: 3.28653338184268
	Free valence electron count: 0
	Valence electron count: 4
- Atom #26
	Sigma charge: 0.03905168747981323
	Pi charge: 0.0
	Total partial charge: 0.03905168747981323
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.412828510668822
	Exerted inductive effect: 1.729007096292111
	Free valence electron count: 0
	Valence electron count: 1
- Atom #27
	Sigma charge: 0.03905168747981323
	Pi charge: 0.0
	Total partial charge: 0.03905168747981323
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.412828510668822
	Exerted inductive effect: 1.729007096292111
	Free valence electron count: 0
	Valence electron count: 1
- Atom #28
	Sigma charge: 0.03905168747981323
	Pi charge: 0.0
	Total partial charge: 0.03905168747981323
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.412828510668822
	Exerted inductive effect: 1.729007096292111
	Free valence electron count: 0
	Valence electron count: 1
- Atom #29
	Sigma charge: -0.013050397513689383
	Pi charge: 0.0
	Total partial charge: -0.013050397513689383
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.86051753902983
	Exerted inductive effect: 3.28653338184268
	Free valence electron count: 0
	Valence electron count: 4
- Atom #30
	Sigma charge: 0.03905168747981323
	Pi charge: 0.0
	Total partial charge: 0.03905168747981323
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.412828510668822
	Exerted inductive effect: 1.729007096292111
	Free valence electron count: 0
	Valence electron count: 1
- Atom #31
	Sigma charge: 0.03905168747981323
	Pi charge: 0.0
	Total partial charge: 0.03905168747981323
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.412828510668822
	Exerted inductive effect: 1.729007096292111
	Free valence electron count: 0
	Valence electron count: 1
- Atom #32
	Sigma charge: 0.03905168747981323
	Pi charge: 0.0
	Total partial charge: 0.03905168747981323
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.412828510668822
	Exerted inductive effect: 1.729007096292111
	Free valence electron count: 0
	Valence electron count: 1
- Atom #33
	Sigma charge: 0.04262999760855917
	Pi charge: 0.0
	Total partial charge: 0.04262999760855917
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.43499348772759
	Exerted inductive effect: 6.629867816987954
	Free valence electron count: 0
	Valence electron count: 1
- Atom #34
	Sigma charge: 0.04262999760855917
	Pi charge: 0.0
	Total partial charge: 0.04262999760855917
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.43499348772759
	Exerted inductive effect: 6.629867816987954
	Free valence electron count: 0
	Valence electron count: 1
- Atom #35
	Sigma charge: 0.027865109356298174
	Pi charge: 0.0
	Total partial charge: 0.027865109356298174
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.343443462364415
	Exerted inductive effect: 10.78820039305899
	Free valence electron count: 0
	Valence electron count: 1
- Atom #36
	Sigma charge: 0.027865109356298174
	Pi charge: 0.0
	Total partial charge: 0.027865109356298174
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.343443462364415
	Exerted inductive effect: 10.78820039305899
	Free valence electron count: 0
	Valence electron count: 1
- Atom #37
	Sigma charge: 0.030006462624431417
	Pi charge: 0.0
	Total partial charge: 0.030006462624431417
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.356736109608883
	Exerted inductive effect: 17.864242110638845
	Free valence electron count: 0
	Valence electron count: 1
- Atom #38
	Sigma charge: 0.030006462624431417
	Pi charge: 0.0
	Total partial charge: 0.030006462624431417
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.356736109608883
	Exerted inductive effect: 17.864242110638845
	Free valence electron count: 0
	Valence electron count: 1
- Atom #39
	Sigma charge: -0.05280657267329697
	Pi charge: 0.004919206898789885
	Total partial charge: -0.047887365774507085
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.642031978720426
	Sigma electronegativity: 8.302053429202296
	Exerted inductive effect: 49.39056539475531
	Free valence electron count: 0
	Valence electron count: 4
- Atom #40
	Sigma charge: -0.026094046678615854
	Pi charge: -0.02191195271622437
	Total partial charge: -0.048005999394840224
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 5.4145026363258095
	Sigma electronegativity: 8.547831642856119
	Exerted inductive effect: 51.342383821652696
	Free valence electron count: 0
	Valence electron count: 4
- Atom #41
	Sigma charge: 0.06462047301854192
	Pi charge: 0.0
	Total partial charge: 0.06462047301854192
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.570893300537143
	Exerted inductive effect: 28.376318796798774
	Free valence electron count: 0
	Valence electron count: 1
- Atom #42
	Sigma charge: 0.062215734320143025
	Pi charge: 0.0
	Total partial charge: 0.062215734320143025
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.556058535503376
	Exerted inductive effect: 28.30371915652937
	Free valence electron count: 0
	Valence electron count: 1
- Atom #43
	Sigma charge: 0.062215734320143025
	Pi charge: 0.0
	Total partial charge: 0.062215734320143025
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.556058535503376
	Exerted inductive effect: 28.303719156529375
	Free valence electron count: 0
	Valence electron count: 1
- Atom #44
	Sigma charge: 0.06462047301854192
	Pi charge: 0.0
	Total partial charge: 0.06462047301854192
	Lone-pair electronegativity: 0.0
	Pi electronegativity: 0.0
	Sigma electronegativity: 7.570893300537143
	Exerted inductive effect: 28.376318796798778
	Free valence electron count: 0
	Valence electron count: 1
